lib/protoflow-blocks/src/blocks/sys/read_socket.rs

```diff
@@ -0,0 +1,240 @@
+extern crate std;
+use crate::prelude::{vec, String};
+use crate::{IoBlocks, StdioConfig, StdioError, StdioSystem, System};
+use protoflow_core::{
+    types::Any, Block, BlockError, BlockResult, BlockRuntime, Message, OutputPort, Port,
+    PortResult, SystemBuilding,
+};
+use protoflow_derive::Block;
+use serde::{Deserialize, Serialize};
+use simple_mermaid::mermaid;
+use std::{
+    format,
+    io::Read,
+    net::{TcpListener, TcpStream},
+    string::ToString,
+    sync::{Arc, Mutex, PoisonError},
+};
+use tracing::{error, info};
+/// A block that reads a proto object from a TCP port.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/sys/read_socket.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     // TODO
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute ReadSocket host="127.0.0.1" port="7077" buffer_size="1024"
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct ReadSocket<T: Message = Any> {
```

Comment:
There is no need for this block to be doing decoding as well. It really should just read bytes (in the form of whatever non-zero-length chunk of bytes is available when attempting to read) from the socket, being on par and compatible with `ReadStdin`, `ReadFile`, and so on.

Reply:
done